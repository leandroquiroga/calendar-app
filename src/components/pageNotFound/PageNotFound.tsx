import React from 'react'
import { useNavigate } from 'react-router-dom';
import logo404 from '../../assets/error404.svg'

export const PageNotFound = () => {

  const navigate = useNavigate();

  const handleNavigate = () => navigate('/', { replace: true });
  return (
    <section
      className='d-flex flex-column justify-content-center align-items-center'
      style={{height: 100 + 'vh'}}
    >
      <article className='container p-2'>
        <img 
          className='d-block mx-auto img-fluid'
          style={{width: 350 + 'px'}}
          src={logo404}
          alt='logo 404'
        />
        <p className='text-center fs-3'>Lo sentimos, no se ha encontrado la URL</p>
      </article>
      <button
        className='btn btn-dark p-2'
        onClick={handleNavigate}
      >
        Regresar
      </button>
    </section>
  )
}
