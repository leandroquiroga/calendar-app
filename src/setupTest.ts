import '@testing-library/jest-dom';
import Enzyme from 'enzyme';
import ReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer,  } from 'enzyme-to-json';

Enzyme.configure({ adapter: new ReactAdapter() });

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}) as any)