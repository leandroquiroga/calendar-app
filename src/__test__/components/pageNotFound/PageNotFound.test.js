import React from 'react'
import { mount } from "enzyme";
import { PageNotFound } from "../../../components/pageNotFound/PageNotFound";
import { Route, Routes, MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom';
import '../../../setupTest';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));


describe('Test on componet </PageNotFound>', () => {

  const wapper = mount(
    <MemoryRouter initialEntries={['/*']}>
      <Routes>
        <Route path='/*' element={
          <PageNotFound />
        }/>
      </Routes >
    </MemoryRouter>
  );    

  test('this test should make a snapshot correctly', () => {
    expect(wapper).toMatchSnapshot();
  });

  test('this test should return to previus page', () => {
    wapper.find('button').prop('onClick')();

    expect(mockNavigate).toHaveBeenCalledWith("/", {"replace": true});
  });
});