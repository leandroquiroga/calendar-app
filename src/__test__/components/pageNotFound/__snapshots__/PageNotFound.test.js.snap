// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on componet </PageNotFound> this test should make a snapshot correctly 1`] = `
<section
  className="d-flex flex-column justify-content-center align-items-center"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <article
    className="container p-2"
  >
    <img
      alt="logo 404"
      className="d-block mx-auto img-fluid"
      src="error404.svg"
      style={
        Object {
          "width": "350px",
        }
      }
    />
    <p
      className="text-center fs-3"
    >
      Lo sentimos, no se ha encontrado la URL
    </p>
  </article>
  <button
    className="btn btn-dark p-2"
    onClick={[Function]}
  >
    Regresar
  </button>
</section>
`;
